src/Catalyst.Core.Modules.Dfs.Tests/IntegrationTests/CoreApi/BitSwapApiTest.cs

```diff
@@ -0,0 +1,528 @@
+                Assert.Equal(0UL, ledger.DataSent);
+                Assert.True(ledger.IsInDebt);
+
+                // TODO: Timing issue here.  ipfsOther could have sent the block
+                // but not updated the stats yet.
+#if false
+                var localPeer = await ipfs.LocalPeer;
+                ledger = await ipfsOther.Bitswap.LedgerAsync(localPeer);
+                Assert.Equal(localPeer, ledger.Peer);
+                Assert.NotEqual(0UL, ledger.BlocksExchanged);
+                Assert.Equal(0UL, ledger.DataReceived);
+                Assert.NotEqual(0UL, ledger.DataSent);
+                Assert.False(ledger.IsInDebt);
+#endif
+            }
+            finally
+            {
+                await _dfsServiceOther.StopAsync();
+                await _dfsService.StopAsync();
+
+                _dfsService.Options.Discovery = new DiscoveryOptions();
+                _dfsServiceOther.Options.Discovery = new DiscoveryOptions();
+
+                (_dfsService.BitSwapService).Protocols = originalProtocols;
+                (_dfsServiceOther.BitSwapService).Protocols = otherOriginalProtocols;
+            }
+        }
+
+        [Fact]
+        public async Task GetsBlock_OnRequest()
+        {
+            _dfsService.Options.Discovery.DisableMdns = true;
+            _dfsService.Options.Discovery.BootstrapPeers = new MultiAddress[0];
+            await _dfsService.StartAsync();
+
+            _dfsServiceOther.Options.Discovery.DisableMdns = true;
+            _dfsServiceOther.Options.Discovery.BootstrapPeers = new MultiAddress[0];
+            await _dfsServiceOther.StartAsync();
+            
+            try
+            {
+                var cts = new CancellationTokenSource(10000);
+                var data = Guid.NewGuid().ToByteArray();
+                var cid = await _dfsServiceOther.BlockApi.PutAsync(data, cancel: cts.Token);
+
+                var remote = _dfsServiceOther.LocalPeer;
+                await _dfsService.SwarmApi.ConnectAsync(remote.Addresses.First(), cts.Token);
+
+                var block = await _dfsService.BlockApi.GetAsync(cid, cts.Token);
+                
+                Assert.Equal(cid, block.Id);
+                Assert.Equal(data, block.DataBytes);
+            }
+            finally
+            {
+                await _dfsServiceOther.StopAsync();
+                await _dfsService.StopAsync();
+                _dfsService.Options.Discovery = new DiscoveryOptions();
+                _dfsServiceOther.Options.Discovery = new DiscoveryOptions();
+            }
+        }
+
+        [Fact]
+        public async Task GetsBlock_CidV1()
+        {
+            await _dfsService.StartAsync();
+            await _dfsServiceOther.StartAsync();
+            try
+            {
+                var data = Guid.NewGuid().ToByteArray();
+                var cid = await _dfsServiceOther.BlockApi.PutAsync(data, "raw"); // @TODO get this from a test prop so we can test against multiple hash algos
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Complete the task associated to this 'TODO' comment.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)